src/components/PeoplePage.tsx

```diff
@@ -1,29 +1,163 @@
+  const centuries = searchParams.getAll('centuries');
+  const query = searchParams.get('query');
+  const sortColumn = searchParams.get('sort') as keyof Person | null;
+  const sortDirection = searchParams.get('order') === 'desc';
+
+  useEffect(() => {
+    setIsLoading(true);
+    getPeople()
+      .then(fetchedPeople => {
+        setPeople(
+          fetchedPeople.map(person => ({
+            ...person,
+            century: person.born ? getCentury(person.born) : undefined,
+          })),
+        );
+      })
+      .catch(() => {
+        setError('Something went wrong');
+      })
+      .finally(() => {
+        setIsLoading(false);
+      });
+  }, []);
+
+  const filteredPeople = useMemo(() => {
+    let filtered = people;
+    const centuriesAsNumbers = centuries.map(Number);
+
+    if (sex) {
+      filtered = filtered.filter(person => person.sex === sex);
+    }
+
+    if (centuries.length > 0) {
+      filtered = filtered.filter(
+        person => person.century && centuriesAsNumbers.includes(person.century),
+      );
+    }
+
+    if (query) {
+      filtered = filtered.filter(person =>
+        person.name.toLowerCase().includes(query.toLowerCase()),
+      );
+    }
+
+    return filtered;
+  }, [people, centuries, query, sex]);
+
+  const sortedFilteredPeople = useMemo(() => {
+    if (!sortColumn) {
+      return filteredPeople;
+    }
+
+    return [...filteredPeople].sort((a, b) => {
+      const valueA = a[sortColumn];
+      const valueB = b[sortColumn];
+
+      if (valueA === undefined) {
+        return 1;
+      }
+
+      if (valueB === undefined) {
+        return -1;
+      }
+
+      if (valueA < valueB) {
+        return sortDirection ? 1 : -1;
+      }
+
+      if (valueA > valueB) {
+        return sortDirection ? -1 : 1;
+      }
+
+      return 0;
+    });
+  }, [filteredPeople, sortColumn, sortDirection]);
+  const handleSearchChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const newParams = new URLSearchParams(searchParams);
+
+    newParams.set('query', event.target.value);
+    setSearchParams(newParams);
+  };
+
+  const peopleMap: PeopleMap = useMemo(
+    () =>
+      people.reduce(
+        (map, person) => ({
+          ...map,
+          [person.name]: person,
+        }),
+        {},
+      ),
+    [people],
+  );
+
+  const toggleArrow = (column: string) => {
+    const [order, sort] =
+      column === sortColumn
+        ? sortDirection
+          ? [null, null]
+          : ['desc', column]
+        : [null, column];
+
+    return { sex, query, centuries, sort, order };
+  };
+
   return (
     <>
       <h1 className="title">People Page</h1>
 
       <div className="block">
         <div className="columns is-desktop is-flex-direction-row-reverse">
           <div className="column is-7-tablet is-narrow-desktop">
-            <PeopleFilters />
+            <PeopleFilters
+              onSearchChange={handleSearchChange}
+              sex={sex ?? undefined}
```

Comment:
What value would _sex_, _centuries_ and _query_ have if when trying to declare them, _searchParams_ didn't have the key you're loking for?
E.g:
```
const sex = searchParam.get('sex');
    // if searchParams = { query: 'batata', centuries: ['15'] }; What would be the value of _sex_ ? 
```